cli/packages/prisma-db-introspection/src/types/graphql.ts

```diff
@@ -0,0 +1,101 @@
+import * as os from 'os'
+import * as _ from 'lodash'
+
+export class GQLType {
+  name: string
+  fields: GQLField[]
+  directives: string[]
+  renderCommented: boolean
+
+  constructor(name: string, fields: GQLField[], directives: string[], renderCommented: boolean) {
+    this.name = name
+    this.fields = fields
+    this.directives = directives
+    this.renderCommented = renderCommented
+  }
+
+  // Determines if this type is valid GraphQL compatible with Prisma
+  isValid(): boolean {
+    return this.fields.some(f => f.isValid())
```

Comment:
Shouldn't this test for `!f.isValid`?

Reply:
If a `GQLType` has any valid fields, it is a valid type. 
Invalid fields are commented out making the rest of the type valid.